File: prog/c/projectile.c
// projectile.c - dragless projectile motion analysis

#include <math.h>
#include <stdio.h>

#define G0 9.80665
#define PI 3.141592654

int main(void) {
	double angle, speed, peak, range, time;

	printf("Angle (deg) = ");
	scanf("%lf", &angle);

	printf("Speed (m/s) = ");
	scanf("%lf", &speed);

	angle *= PI / 180;

	peak  = pow(speed * sin(angle), 2) / (2 * G0);
	range = pow(speed, 2) * sin(2 * angle) / G0;
	time  = 2 * speed * sin(angle) / G0;

	printf("Peak  (m)   = %g\n", peak);
	printf("Range (m)   = %g\n", range);
	printf("Time  (s)   = %g\n", time);
}

File: prog/python/projectile.py
#! /usr/bin/env python3

# projectile.py - dragless projectile motion analysis

from math import radians, sin

G0 = 9.80665

angle = float(input('Angle (deg) = '))
speed = float(input('Speed (m/s) = '))

angle = radians(angle)

print(f'Peak  (m)   = {pow(speed * sin(angle), 2) / 2 / G0:g}')
print(f'Range (m)   = {pow(speed, 2) * sin(2 * angle) / G0:g}')
print(f'Time  (s)   = {2 * speed * sin(angle) / G0:g}')
